### PR TITLE
Add info about Mac Catalyst and update samples

### PR DESCRIPTION
**Pull request by cardoso**
- Added Mac Catalyst target and missing entitlements in samples
- Updated README with info
- Doesn't work with Carthage for lack of xcframework support ([ref](https://github.com/Carthage/Carthage/issues/2799))

#no_changelog

### CODE REVIEW

File: Example/Carthage/ChatExample.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -564,8 +579,8 @@
 				ASSETCATALOG_COMPILER_APPICON_NAME = AppIcon;
 				CLANG_ENABLE_MODULES = YES;
 				CODE_SIGN_ENTITLEMENTS = ChatExample/ChatExample.entitlements;
-				CODE_SIGN_IDENTITY = "Apple Development: Bahadir Oncel (C42UH28Y58)";
-				CODE_SIGN_STYLE = Manual;
+				CODE_SIGN_IDENTITY = "Apple Development";
+				CODE_SIGN_STYLE = Automatic;
```
Comment (b-onc) [inline Example/Carthage/ChatExample.xcodeproj/project.pbxproj:583]: You shouldn't touch this part

File: Example/Carthage/ChatExample.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -577,7 +592,7 @@
 				);
 				PRODUCT_BUNDLE_IDENTIFIER = io.getstream.ChatExample;
 				PRODUCT_NAME = "$(TARGET_NAME)";
-				PROVISIONING_PROFILE_SPECIFIER = "match Development io.getstream.ChatExample";
+				PROVISIONING_PROFILE_SPECIFIER = "";
```
Comment (b-onc) [inline Example/Carthage/ChatExample.xcodeproj/project.pbxproj:595]: You shouldn't touch this part

File: Example/Cocoapods/ChatExample.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -373,7 +373,8 @@
 			buildSettings = {
 				ASSETCATALOG_COMPILER_APPICON_NAME = AppIcon;
 				CODE_SIGN_ENTITLEMENTS = ChatExample/ChatExample.entitlements;
-				CODE_SIGN_STYLE = Manual;
+				CODE_SIGN_IDENTITY = "Apple Development";
+				CODE_SIGN_STYLE = Automatic;
```
Comment (b-onc) [inline Example/Cocoapods/ChatExample.xcodeproj/project.pbxproj:377]: You shouldn't touch this part

File: Example/Cocoapods/ChatExample.xcodeproj/project.pbxproj [hunk 4]
```diff
@@ -385,7 +386,8 @@
 				MARKETING_VERSION = 1.3.12;
 				PRODUCT_BUNDLE_IDENTIFIER = io.getstream.ChatExample;
 				PRODUCT_NAME = "$(TARGET_NAME)";
-				PROVISIONING_PROFILE_SPECIFIER = "match Development io.getstream.ChatExample";
+				PROVISIONING_PROFILE_SPECIFIER = "";
```
Comment (b-onc) [inline Example/Cocoapods/ChatExample.xcodeproj/project.pbxproj:389]: You shouldn't touch this part

File: Example/Carthage/ChatExample.xcodeproj/project.pbxproj (hunk 1 above)
Comment (cardoso) [inline Example/Carthage/ChatExample.xcodeproj/project.pbxproj:583]:
```suggestion
				CODE_SIGN_IDENTITY = "Apple Development: Bahadir Oncel (C42UH28Y58)";
				CODE_SIGN_STYLE = Manual;
```

File: Example/Carthage/ChatExample.xcodeproj/project.pbxproj (hunk 2 above)
Comment (cardoso) [inline Example/Carthage/ChatExample.xcodeproj/project.pbxproj:595]:
```suggestion
				PROVISIONING_PROFILE_SPECIFIER = "match Development io.getstream.ChatExample";
```

File: Example/Cocoapods/ChatExample.xcodeproj/project.pbxproj (hunk 3 above)
Comment (cardoso) [inline Example/Cocoapods/ChatExample.xcodeproj/project.pbxproj:377]:
```suggestion
				CODE_SIGN_IDENTITY = Manual;
```

File: Example/Cocoapods/ChatExample.xcodeproj/project.pbxproj (hunk 4 above)
Comment (cardoso) [inline Example/Cocoapods/ChatExample.xcodeproj/project.pbxproj:389]:
```suggestion
				PROVISIONING_PROFILE_SPECIFIER = "match Development io.getstream.ChatExample";
```

File: Example/Cocoapods/ChatExample.xcodeproj/project.pbxproj [hunk 5]
```diff
@@ -373,7 +373,7 @@
 			buildSettings = {
 				ASSETCATALOG_COMPILER_APPICON_NAME = AppIcon;
 				CODE_SIGN_ENTITLEMENTS = ChatExample/ChatExample.entitlements;
-				CODE_SIGN_STYLE = Manual;
+				CODE_SIGN_IDENTITY = Manual;
```
Comment (cardoso) [inline Example/Cocoapods/ChatExample.xcodeproj/project.pbxproj:376]:
```suggestion
				CODE_SIGN_STYLE = Manual;
```

**Comment by cardoso** [comment]
@b-onc removed the code signing changes & fixed conflicts.

Btw, why can't we leave code signing automatic in the samples?

File: README.md [hunk 6]
```diff
@@ -43,12 +43,18 @@ You'll typically want to start out using the UI components, and implement your c
...
-- iOS 11+
+- iOS 11+/macOS 10.15+
 - Xcode 11.2+
 - Swift 5.1
 - CocoaPods 1.7+
 - Carthage 0.33.0+
 
+## Mac Catalyst
+
+Our frameworks are compatible with [Mac Catalyst](https://developer.apple.com/mac-catalyst/), meaning you can create a chat application that works on both iOS and macOS. Read about it [here](link-to-blog-post).
```
Comment (cardoso) [inline README.md:54]:
```suggestion
Our frameworks are compatible with [Mac Catalyst](https://developer.apple.com/mac-catalyst/), meaning you can create a chat application that works on both iOS and macOS. Read about it [here](https://getstream.io/blog/stream-chat-swift-macos-support).
```

**Pull request merged**